pkg/local_object_storage/metabase/select.go

```diff
@@ -223,10 +212,57 @@ func (db *DB) selectFastFilter(
 		selectAllFromBucket(tx, storageGroupBucketName(cid), prefix, to, fNum)
 	default: // user attribute
 		bucketName := attributeBucketName(cid, f.Header())
-		db.selectFromFKBT(tx, bucketName, f, prefix, to, fNum)
+
+		if f.Operation() == object.MatchNotPresent {
+			selectOutsideFKBT(tx, allBucketNames(cid), bucketName, f, prefix, to, fNum)
+		} else {
+			db.selectFromFKBT(tx, bucketName, f, prefix, to, fNum)
+		}
 	}
 }
 
+// TODO: move to DB struct
+var mBucketNaming = map[string][]func(*container.ID) []byte{
+	v2object.TypeRegular.String():      {primaryBucketName, parentBucketName},
+	v2object.TypeTombstone.String():    {tombstoneBucketName},
+	v2object.TypeStorageGroup.String(): {storageGroupBucketName},
+}
+
+func allBucketNames(cid *container.ID) (names [][]byte) {
+	for _, fns := range mBucketNaming {
+		for _, fn := range fns {
+			names = append(names, fn(cid))
+		}
+	}
+
+	return
+}
+
+func bucketNamesForType(cid *container.ID, mType object.SearchMatchType, typeVal string) (names [][]byte) {
```

Comment:
Maybe `allBucketNames` can be reused in this function if we pass `object.MatchNotPresent` type. Just a thought.

Reply:
I thought about that, but I decided that `allBucketNames` could have simpler logic, which is necessary when processing `MatchNotPresent`. Call `bucketNamesForType(cid, object.MatchNotPresent, "")` will not give the desired result.